### ISSUE TITLE
VFS Module broken after update to 0.15.6

### ISSUE THREAD

**Issue by danielrivard** [gboudreau/Greyhole#298]
It seems that the latest update may have messed up the installed vfs module.

Ubuntu Server 20.04.4 LTS
Samba 4.13
Greyhole 0.15.16

Executed build_vfs.sh, rebooted, problem fixed. But I thought you should know.

```
May 20 18:59:44 INFO daemon: Greyhole (version 0.15.16) daemon started.
May 20 18:59:44 INFO daemon: Checking MySQL tables...
May 20 18:59:44 DEBUG daemon: Loading metadata store backup directories...
May 20 18:59:44 DEBUG daemon:   Found 2 directories in the settings table.
May 20 18:59:45 DEBUG daemon: Checking symlink at /usr/lib/x86_64-linux-gnu/samba/vfs/greyhole.so...
May 20 18:59:45 DEBUG daemon:   Is OK.
May 20 18:59:45 WARN daemon:   Greyhole VFS module (/usr/lib/x86_64-linux-gnu/samba/vfs/greyhole.so) seems to be missing some required libraries. If you have issues connecting to your Greyhole-enabled shares, try 
to compile a new VFS module for Samba by running this command: sudo /usr/share/greyhole/build_vfs.sh current
```

```
[2022/05/20 19:01:46.744829,  0] ../../lib/util/modules.c:49(load_module)
  Error loading module '/usr/lib/x86_64-linux-gnu/samba/vfs/greyhole.so': libsmbd-conn-samba4.so: cannot open shared object file: No such file or directory
[2022/05/20 19:01:46.744886,  0] ../../source3/smbd/vfs.c:185(vfs_init_custom)
  error probing vfs module 'greyhole': NT_STATUS_UNSUCCESSFUL
[2022/05/20 19:01:46.744916,  0] ../../source3/smbd/vfs.c:397(smbd_vfs_init)
  smbd_vfs_init: vfs_init_custom failed for greyhole
[2022/05/20 19:01:46.744934,  0] ../../source3/smbd/service.c:636(make_connection_snum)
  make_connection_snum: vfs_init failed for service Backup
```

**Comment by gboudreau** [comment]
Hey, Dan! My man! ;)

Can you `ldd /usr/lib/x86_64-linux-gnu/samba/vfs/greyhole.so | egrep 'smb|samba`, and paste in a new comment what you get?

I build the new VFS modules in Ubuntu Docker images... So I'm not sure how you build and mine differs.

**Comment by danielrivard** [comment]
```
~# ldd /usr/lib/x86_64-linux-gnu/samba/vfs/greyhole.so | egrep 'smb|samba'
	libsamba-util.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libsamba-util.so.0 (0x00007f4e82c01000)
	libsmbd-base-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsmbd-base-samba4.so (0x00007f4e8281c000)
	libtalloc.so.2 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libtalloc.so.2 (0x00007f4e82810000)
	libtevent.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libtevent.so.0 (0x00007f4e827f6000)
	libsamba-debug-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsamba-debug-samba4.so (0x00007f4e827ea000)
	libsmbconf.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libsmbconf.so.0 (0x00007f4e82738000)
	libtevent-util.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libtevent-util.so.0 (0x00007f4e82731000)
	libgenrand-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libgenrand-samba4.so (0x00007f4e82357000)
	libsocket-blocking-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsocket-blocking-samba4.so (0x00007f4e82352000)
	libsys-rw-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsys-rw-samba4.so (0x00007f4e8234b000)
	libtime-basic-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libtime-basic-samba4.so (0x00007f4e82346000)
	libreplace-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libreplace-samba4.so (0x00007f4e82341000)
	libndr.so.1 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libndr.so.1 (0x00007f4e822fb000)
	liblibsmb-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/liblibsmb-samba4.so (0x00007f4e82272000)
	libdbwrap-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libdbwrap-samba4.so (0x00007f4e82263000)
	libtdb.so.1 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libtdb.so.1 (0x00007f4e82245000)
	libsamba-security-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsamba-security-samba4.so (0x00007f4e8221c000)
	libndr-samba-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libndr-samba-samba4.so (0x00007f4e81ff1000)
	libkrb5samba-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libkrb5samba-samba4.so (0x00007f4e81fdd000)
	libmessages-dgm-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libmessages-dgm-samba4.so (0x00007f4e81fcc000)
	libprinting-migrate-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libprinting-migrate-samba4.so (0x00007f4e81fad000)
	libdcerpc-server-core.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libdcerpc-server-core.so.0 (0x00007f4e81f94000)
	libsamba-sockets-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsamba-sockets-samba4.so (0x00007f4e81f76000)
	libcliauth-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcliauth-samba4.so (0x00007f4e81f58000)
	libnpa-tstream-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libnpa-tstream-samba4.so (0x00007f4e81f4c000)
	libauth-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libauth-samba4.so (0x00007f4e81f1d000)
	libcommon-auth-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcommon-auth-samba4.so (0x00007f4e81f12000)
	libsamba3-util-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsamba3-util-samba4.so (0x00007f4e81f03000)
	libserver-id-db-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libserver-id-db-samba4.so (0x00007f4e81efb000)
	libsamba-passdb.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libsamba-passdb.so.0 (0x00007f4e81eaa000)
	libsamba-errors.so.1 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libsamba-errors.so.1 (0x00007f4e81d53000)
	libiov-buf-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libiov-buf-samba4.so (0x00007f4e81d4e000)
	libmsrpc3-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libmsrpc3-samba4.so (0x00007f4e81d23000)
	libnetapi.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libnetapi.so.0 (0x00007f4e81c90000)
	libCHARSET3-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libCHARSET3-samba4.so (0x00007f4e81c8a000)
	libsamba-cluster-support-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsamba-cluster-support-samba4.so (0x00007f4e81c85000)
	libdcerpc-samba-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libdcerpc-samba-samba4.so (0x00007f4e81ad2000)
	libutil-tdb-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libutil-tdb-samba4.so (0x00007f4e81acc000)
	libgse-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libgse-samba4.so (0x00007f4e81a97000)
	libcli-smb-common-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcli-smb-common-samba4.so (0x00007f4e81a53000)
	libgensec-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libgensec-samba4.so (0x00007f4e81a18000)
	libtdb-wrap-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libtdb-wrap-samba4.so (0x00007f4e81a13000)
	libndr-standard.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libndr-standard.so.0 (0x00007f4e81486000)
	libcom_err-samba4.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcom_err-samba4.so.0 (0x00007f4e81481000)
	libcli-spoolss-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcli-spoolss-samba4.so (0x00007f4e81476000)
	libutil-reg-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libutil-reg-samba4.so (0x00007f4e81471000)
	libsmbd-shim-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsmbd-shim-samba4.so (0x00007f4e8146c000)
	libsamba-hostconfig.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libsamba-hostconfig.so.0 (0x00007f4e81436000)
	libdcerpc-binding.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libdcerpc-binding.so.0 (0x00007f4e81403000)
	libutil-setid-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libutil-setid-samba4.so (0x00007f4e813fc000)
	liblibcli-lsa3-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/liblibcli-lsa3-samba4.so (0x00007f4e813f6000)
	libads-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libads-samba4.so (0x00007f4e813e4000)
	libsamba-modules-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsamba-modules-samba4.so (0x00007f4e813df000)
	libwbclient.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libwbclient.so.0 (0x00007f4e813c9000)
	libsamba-credentials.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libsamba-credentials.so.0 (0x00007f4e813b0000)
	libsecrets3-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsecrets3-samba4.so (0x00007f4e8138d000)
	libndr-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libndr-samba4.so (0x00007f4e81061000)
	libauthkrb5-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libauthkrb5-samba4.so (0x00007f4e81042000)
	libmessages-util-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libmessages-util-samba4.so (0x00007f4e81035000)
	libtalloc-report-printf-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libtalloc-report-printf-samba4.so (0x00007f4e81030000)
	libinterfaces-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libinterfaces-samba4.so (0x00007f4e8102a000)
	libserver-role-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libserver-role-samba4.so (0x00007f4e81025000)
	libasn1util-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libasn1util-samba4.so (0x00007f4e80bfa000)
	libutil-cmdline-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libutil-cmdline-samba4.so (0x00007f4e80bf3000)
	libsmb-transport-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsmb-transport-samba4.so (0x00007f4e80bea000)
	libndr-nbt.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libndr-nbt.so.0 (0x00007f4e80ba7000)
	libasn1-samba4.so.8 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libasn1-samba4.so.8 (0x00007f4e80af8000)
	libkrb5-samba4.so.26 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libkrb5-samba4.so.26 (0x00007f4e80a68000)
	libgssapi-samba4.so.2 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libgssapi-samba4.so.2 (0x00007f4e80a21000)
	libmsghdr-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libmsghdr-samba4.so (0x00007f4e80a1a000)
	libdcerpc-pkt-auth-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libdcerpc-pkt-auth-samba4.so (0x00007f4e80a15000)
	liblibcli-netlogon3-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/liblibcli-netlogon3-samba4.so (0x00007f4e80a0b000)
	libMESSAGING-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libMESSAGING-samba4.so (0x00007f4e809fe000)
	libsamdb.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libsamdb.so.0 (0x00007f4e809db000)
	libndr-krb5pac.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/libndr-krb5pac.so.0 (0x00007f4e809c3000)
	libflag-mapping-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libflag-mapping-samba4.so (0x00007f4e80983000)
	libtrusts-util-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libtrusts-util-samba4.so (0x00007f4e80978000)
	libaddns-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libaddns-samba4.so (0x00007f4e80968000)
	libcli-cldap-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcli-cldap-samba4.so (0x00007f4e8095c000)
	libcli-nbt-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcli-nbt-samba4.so (0x00007f4e8094d000)
	libwinbind-client-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libwinbind-client-samba4.so (0x00007f4e80945000)
	libldbsamba-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libldbsamba-samba4.so (0x00007f4e8090c000)
	libldb.so.2 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libldb.so.2 (0x00007f4e808c9000)
	libsamdb-common-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libsamdb-common-samba4.so (0x00007f4e8088f000)
	libroken-samba4.so.19 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libroken-samba4.so.19 (0x00007f4e8086e000)
	libhx509-samba4.so.5 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libhx509-samba4.so.5 (0x00007f4e8080a000)
	libheimbase-samba4.so.1 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libheimbase-samba4.so.1 (0x00007f4e80801000)
	libwind-samba4.so.0 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libwind-samba4.so.0 (0x00007f4e807d5000)
	libhcrypto-samba4.so.5 => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libhcrypto-samba4.so.5 (0x00007f4e80783000)
	libMESSAGING-SEND-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libMESSAGING-SEND-samba4.so (0x00007f4e8077e000)
	libcli-ldap-common-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libcli-ldap-common-samba4.so (0x00007f4e80771000)
	libclidns-samba4.so => /usr/share/greyhole/vfs-build/samba-4.13.17/bin/shared/private/libclidns-samba4.so (0x00007f4e80766000)
```

**Comment by gboudreau** [comment]
0.15.17 should resolve this.

VFS module for Samba 4.13 should be compiled with the most recent available minor version of Samba; building using 4.13.0 creates a dependency on a shared library that is not available without compiling (libsmbd-conn-samba4.so); also recompiled VFS modules for 4.15, 4.14, 4.12 & 4.11 with the latest minor versions of those too, just in case.
